sds/src/match_validation/validator_utils.rs

```diff
@@ -14,7 +14,7 @@ pub fn new_match_validator_from_type(config: &MatchValidationType) -> Box<dyn Ma
             _ => panic!("This aws type shall not be used to create a validator"),
         },
         MatchValidationType::CustomHttp(http_config) => {
-            Box::new(HttpValidator::new(http_config.clone()))
+            Box::new(HttpValidator::new_clone_config(http_config))
```

Comment:
I did not realised that in the previous PRs, but this can be moved directly in the `impl MatchValidationType` with a method like `pub fn into_match_validator(self) -> Box<dyn MatchValidator> {`, and so then the code in the scanner will be `match_validation_type.clone().into_match_validator()`. This will reduce the number of imported files. 
It can be tackled in a follow-up PR as it is not directly related to the current change.

Reply:
I fixed that in this PR it is fair enouh